app/src/main/java/com/microsoft/research/karya/ui/views/KaryaToolbar.kt

```diff
@@ -86,6 +87,15 @@ class KaryaToolbar : Toolbar {
         }
     }
 
+    fun setBackBtnClickListener(onClick: OnClickListener) {
+        binding.toolbarBackBtn.apply {
+            visible() // doesn't makes sense to set click-listener on an invisible view
+            setOnClickListener(onClick)
+        }
+    }
+
+    fun hideBackButton(): Unit = binding.toolbarBackBtn.invisible()
+
     fun setProfileClickListener(onClick: () -> Unit) {
         binding.profilePictureIv.apply {
             visible()
```

Comment:
Good work on migrating all of that! Could you also remove the dependency on kotlinx synthetic from gradle

Reply:
actually not everything is migrated, I was having issues with testing the SpeechTranscription, it was throwing `corrupt audio file`, I used the assets from `speech verification` sample. Would be done by today, three scenarios remaining.